### PR TITLE
item text is now editable input

### PR DESCRIPTION
**Pull request by Joseph-Cashmore**
I'm not 100% that I have gone about this in the correct/most efficient way.

The text in the items is now an input field, that the user can click on and change.

I have added some css to align with the designs.

### CODE REVIEW

File: src/components/ShoppingList/ShoppingList.js [hunk 1]
```diff
@@ -29,13 +29,28 @@ const ShoppingList = () => {
     setItems(filteredItems);
   };
 
+  const updateItem = (edit, key) => {
+    const updatedItems = items.map((item) => {
+      const newItem = item;
+      if (item.key === key) {
+        newItem.text = edit;
```
Comment (samstradling) [inline src/components/ShoppingList/ShoppingList.js]: It's best to not mutate the state directly and just return the new value, i.e. change this line to:
```suggestion
        return {
          ...item,
          text: edit,
        };
```

File: src/components/ShoppingList/ShoppingList.js [hunk 2]
```diff
@@ -29,13 +29,28 @@ const ShoppingList = () => {
     setItems(filteredItems);
   };
 
+  const updateItem = (edit, key) => {
```
Comment (samstradling) [inline src/components/ShoppingList/ShoppingList.js]: Maybe this is clearer?
```suggestion
  const updateItem = (updatedValue, key) => {
```

File: debug.log [hunk 3]
```diff
@@ -0,0 +1 @@
+[1103/115006.647:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
```
Comment (samstradling) [inline debug.log]: I guess you didn't mean to commit this?

**Comment by samstradling** [comment]
Looking great! 😄

File: debug.log (hunk 3 above)
Comment (Joseph-Cashmore) [reply debug.log]: nope!

File: src/components/ListItems/ListItems.js [hunk 4]
```diff
@@ -4,11 +4,16 @@ import { faTrash } from '@fortawesome/free-solid-svg-icons';
...
   <ul className={styles.list}>
     {entries.map(({ key, text }) => (
       <li key={key} className={styles.listItem}>
-        {text}
+        <input
+          type="text"
+          id={key}
+          value={text}
+          onChange={(e) => updateItem(e.target.value, key)}
+        />
```
Comment (samstradling) [inline src/components/ListItems/ListItems.js:17]: So how about some tests? 😄

**Review by samstradling — APPROVED** [review]
Looks great!

**Pull request merged**